components/patient-data/rest/src/test/java/org/phenotips/data/rest/internal/DefaultDomainObjectFactoryTest.java

```diff
@@ -0,0 +1,260 @@
+import org.xwiki.model.reference.DocumentReferenceResolver;
+import org.xwiki.security.authorization.AuthorizationManager;
+import org.xwiki.security.authorization.Right;
+import org.xwiki.test.mockito.MockitoComponentMockingRule;
+import org.xwiki.users.UserManager;
+
+import javax.inject.Named;
+import javax.ws.rs.core.UriBuilder;
+import javax.ws.rs.core.UriInfo;
+import java.lang.reflect.ParameterizedType;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+import static org.mockito.Mockito.*;
+
+public class DefaultDomainObjectFactoryTest {
+
+    @Rule
+    public MockitoComponentMockingRule<DomainObjectFactory> mocker =
+            new MockitoComponentMockingRule<DomainObjectFactory>(DefaultDomainObjectFactory.class);
+
+    @Mock
+    private Patient patient;
+
+    @Mock
+    private UriInfo uriInfo;
+
+    private ParameterizedType stringResolverType = new DefaultParameterizedType(null, DocumentReferenceResolver.class,
+            String.class);
+
+    private AuthorizationManager access;
+
+    private UserManager users;
+
+    private DocumentAccessBridge documentAccessBridge;
+
+    @Named("current")
+    private DocumentReferenceResolver<String> stringResolver;
+
+    @Before
+    public void setUp() throws ComponentLookupException {
+        MockitoAnnotations.initMocks(this);
+        this.access = this.mocker.getInstance(AuthorizationManager.class);
+        this.users = this.mocker.getInstance(UserManager.class);
+        this.documentAccessBridge = this.mocker.getInstance(DocumentAccessBridge.class);
+        this.stringResolver = this.mocker.getInstance(this.stringResolverType, "current");
+
+        when(this.users.getCurrentUser()).thenReturn(null);
+    }
+
+    @Test
+    public void cannotCreatePatientWithNoAccess() throws ComponentLookupException {
+        when(this.patient.getDocument()).thenReturn(null);
+        when(this.access.hasAccess(Right.VIEW, null, null)).thenReturn(false);
+        assertNull(this.mocker.getComponentUnderTest().createPatientSummary(this.patient, this.uriInfo));
+    }
+
+    @Test
+    public void createPatientDocumentExceptionReturnsNull() throws Exception {
+        when(this.patient.getDocument()).thenReturn(null);
+        when(this.access.hasAccess(Right.VIEW, null, null)).thenReturn(true);
+
+        doThrow(Exception.class).when(this.documentAccessBridge).getDocument(Matchers.any(DocumentReference.class));
+        assertNull(this.mocker.getComponentUnderTest().createPatientSummary(this.patient, this.uriInfo));
+    }
+
+    @Test
+    public void createPatientPerformsCorrectly() throws ComponentLookupException, URISyntaxException, Exception
+    {
+        XWikiDocument documentReference = mock(XWikiDocument.class);
+        DocumentReference patientDocument = mock(DocumentReference.class);
```

Comment:
`DocumentReference` are very stupid POJOs, you shouldn't mock them but instantiate a new one:

    new DocumentReference("wiki", "data", "P0000001")